### PR TITLE
Disable Power Down option for desk devices

### PR DESCRIPTION
**Pull request by geohot**
Non UI parameter.

cd /data/openpilot/common
'./params.py DisablePowerDown 1' to use

### CODE REVIEW

**Review by pd0wm — APPROVED** [review]

File: common/params.py [hunk 1]
```diff
@@ -407,10 +409,10 @@ def f(key, val):
       pp = params.get(k)
       if pp is None:
         print("%s is None" % k)
-      elif all(ord(c) < 128 and ord(c) >= 32 for c in pp):
+      elif all(c < 128 and c >= 32 for c in pp):
```
Comment (pd0wm) [inline common/params.py]: could use `chr(c) in string.printable`

File: selfdrive/boardd/boardd.cc [hunk 2]
```diff
@@ -392,12 +392,21 @@ void can_health(PubSocket *publisher) {
...
+      // reset the counter to not continually poll the parameter
+      no_ignition_cnt = 0;
+    } else {
+      printf("TURN OFF CHARGING!\n");
+      pthread_mutex_lock(&usb_lock);
+      libusb_control_transfer(dev_handle, 0xc0, 0xe6, (uint16_t)(cereal::HealthData::UsbPowerMode::CLIENT), 0, NULL, 0, TIMEOUT);
+      pthread_mutex_unlock(&usb_lock);
+      printf("POWER DOWN DEVICE\n");
+      system("service call power 17 i32 0 i32 1");
+    }
```
Comment (pd0wm) [inline selfdrive/boardd/boardd.cc]: Where do you free `disbale_power_down`?

**Comment by geohot** [comment]
Fixed free, in general, we should move code like that to C++. Was stupidly thinking it would go out of scope.

Also, will power down right away if the ignition timer is expired and you clear the param. The polling shouldn't be expensive.

File: common/params.py (hunk 1 above)
Comment (damianzim) [reply common/params.py]: `string.printable` returns some (white) ASCII characters with value under 32

**Pull request merged**